scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,351 @@
+    }
+
+    val dependenciesWithoutAnyLinks = getDependenciesThatNeedIntervention(finalDependenciesList)
+    if (dependenciesWithoutAnyLinks.isNotEmpty()) {
+      println(
+        """
+        Please remove all the invalid links (if any) for the below mentioned dependencies
+        and provide the valid license links manually:
+        """.trimIndent()
+      )
+      dependenciesWithoutAnyLinks.forEach { dependency ->
+        println(dependency)
+      }
+      throw Exception("License links are invalid or not available for some dependencies")
+    }
+    println("\nScript executed succesfully: maven_dependencies.textproto updated successfully.")
+  }
+
+  private fun addChangesFromTextProto(
+    dependencyListFromPom: List<MavenDependency>,
+    dependencyListFromProto: List<MavenDependency>
+  ): List<MavenDependency> {
+    return dependencyListFromPom.map { dependency ->
+      dependencyListFromProto.find {
+        it.artifactName == dependency.artifactName
+      } ?: dependency
+    }
+  }
+
+  private fun retrieveManuallyUpdatedLicensesSet(
+    mavenDependenciesList: List<MavenDependency>
+  ): Set<License> {
+    return mavenDependenciesList.flatMap { dependency ->
+      dependency.licenseList.filter { license ->
+        !license.verifiedLinkCase.equals(License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET)
+      }
+    }.toSet()
+  }
+
+  private fun getDependenciesThatNeedIntervention(
+    mavenDependenciesList: List<MavenDependency>
+  ): Set<MavenDependency> {
+    return mavenDependenciesList.filter { dependency ->
+      dependency.licenseList.isEmpty() ||
+        dependency.licenseList.filter { license ->
+          license.verifiedLinkCase.equals(License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET) &&
+            license.isOriginalLinkInvalid == true
+        }.isNotEmpty()
+    }.toSet()
+  }
+
+  private fun getAllBrokenLicenses(
+    mavenDependenciesList: List<MavenDependency>
+  ): Set<License> {
+    return mavenDependenciesList.flatMap { dependency ->
+      dependency.licenseList.filter { license ->
+        license.verifiedLinkCase.equals(License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET) &&
+          license.isOriginalLinkInvalid == false
+      }
+    }.toSet()
+  }
+
+  private fun updateMavenDependenciesList(
+    latestDependenciesList: List<MavenDependency>,
+    manuallyUpdatedLicenses: Set<License>
+  ): List<MavenDependency> {
+    val finalUpdatedList = mutableListOf<MavenDependency>()
+    latestDependenciesList.forEach { mavenDependency ->
+      val updateLicenseList = mutableListOf<License>()
+      mavenDependency.licenseList.forEach { license ->
+        val updatedLicense = manuallyUpdatedLicenses.find {
+          it.originalLink == license.originalLink && it.licenseName == license.licenseName
+        }
+        if (updatedLicense != null) {
+          updateLicenseList.add(updatedLicense)
+        } else {
+          updateLicenseList.add(license)
+        }
+      }
```

Comment:
```suggestion
      val updateLicenseList = mavenDependency.licenseList.map { license ->
        manuallyUpdatedLicenses.find {
          it.originalLink == license.originalLink && it.licenseName == license.licenseName
        } ?: license
      }
```

Shorter & more idiomatic.

Reply:
Done.